### ISSUE TITLE
vui does not run outside CUDA, and the published checkpoint does not match the cod

### ISSUE THREAD

**Issue by robdmac** [fluxions-ai/vui#27]
Trying to run TTS inference on an M2 (vui 1.0.0, main @ a3c38b99c29a, python 3.12.8, torch 2.11.0).
I started on the MLX path, but neither problem below is MLX-specific.

**1. `Engine` is CUDA-only in two places.** `__init__` takes no `device` argument and
`engine.py:314` calls `.cuda()` unconditionally, so the README example dies with
`Torch not compiled with CUDA enabled`. A CPU-only Linux install fails the same way. Passing
`model=`/`codec=` ready-built gets past that, then `__init__` calls `_setup_graphs()` unguarded at
line 349, which hits `setup_cuda_graph_kv` at line 420. There is no route to a working `Engine`
without CUDA.

**2. The published checkpoint does not bind.** Loading `ckpts-vui-cohost-100m.pt` through the
native torch loader drops 18 keys:

```
Ignoring parameter 'audio_embeddings.{0..8}.weight' because it is not found in the model
Ignoring parameter 'audio_heads.{0..8}.weight' because it is not found in the model
```

That is the whole audio embedding and head stack for all 9 quantizers, leaving
`model.rq_transformer` as `None`. The MLX loader fails on the same mismatch, so it is checkpoint
against code, not a backend bug. `vui-190k.pt` and `ckpts-vui-190k.pt` both 404; the README
describes a 300M Nano; the only published `.pt` is the 100M cohost one.

**Which checkpoint is the code written against, and is it published?**

**3. Minor.** `mlx/tts/weights.py:85` fails with `Can't call numpy() on Tensor that requires grad`.
`t.float()` should be `t.detach().float()`. Changes no values, and blocks MLX users before they
reach issue 2.

Happy to test a fix or supply full tracebacks

**Comment by mogwai** [comment]
Thanks for the detailed report — all three points were real, and everything is now fixed on main (#28).

To answer the headline question: the code is written against **`vui-190k.safetensors`** (the default) and `vui-nano.safetensors`, both published at https://huggingface.co/fluxions/vui. The `*100m.pt` files (`ckpts-vui-cohost-100m.pt` etc.) are legacy checkpoints from the original release architecture — per-quantizer `audio_embeddings`/`audio_heads`, no rq_transformer — which is exactly the 18-key mismatch you hit. Both loaders now reject them with an error saying this instead of half-loading with `rq_transformer=None`.

On your other points, as of #28:

1. `Engine` is intentionally CUDA-only (it's built on CUDA graphs), but it now raises a clear error up front instead of dying inside `.cuda()`, and the README example is marked accordingly. On Apple Silicon, `python demo.py --render "some text"` now works too — it previously exited into the CUDA CLI before the MLX auto-detection ran; it now routes to an MLX one-shot render (verified on M-series, ~1.2–2.7× real-time).
2. See above — use the safetensors checkpoints; the loader will now tell you if you grab a legacy one.
3. `torch_to_mlx` now detaches before `.numpy()` — your legacy checkpoint carries `requires_grad` on every tensor, which reproduced the error exactly.

So on your M2: pull main and run `python demo.py` (Gradio UI) or `python demo.py --render --text "..."` — the right weights download automatically. Thanks again for testing, and let us know if anything else comes up on the MLX path.
